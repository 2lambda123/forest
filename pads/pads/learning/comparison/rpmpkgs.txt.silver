Starting on files data/rpmpkgs.txt
886 records.

Before reduction:
Punion(Id = BTy_25 886)
	Pstruct(Id = BTy_24 885)
		Punion(Id = BTy_1 885)
			[Empty] (Id = BTy_0 876);
			Pstruct(Id = BTy_21 9)
				[String] (Id = BTy_2 9);
				Punion(Id = BTy_16 9)
					Pstruct(Id = BTy_15 8)
						[Other](+) (Id = BTy_4 8);
						Punion(Id = BTy_12 8)
							Pstruct(Id = BTy_11 6)
								[Other](+) (Id = BTy_5 6);
								[Pint] (Id = BTy_7 6);
								Punion(Id = BTy_9 6)
									[Empty] (Id = BTy_8 4);
									[Pint] (Id = BTy_10 2);
								End Punion;
							End Pstruct;
							[Pint] (Id = BTy_13 2);
						End Punion;
						[Other](.) (Id = BTy_14 8);
					End Pstruct;
					Pstruct(Id = BTy_20 1)
						[Other](+) (Id = BTy_17 1);
						[Other](+) (Id = BTy_18 1);
						[Other](-) (Id = BTy_19 1);
					End Pstruct;
				End Punion;
			End Pstruct;
		End Punion;
		[Host] (Id = BTy_22 885);
		[String] (Id = BTy_23 885);
	End Pstruct;
	Pstruct(Id = BTy_34 1)
		[String] (Id = BTy_26 1);
		[Other](.) (Id = BTy_27 1);
		[Other](() (Id = BTy_28 1);
		[String] (Id = BTy_29 1);
		[Other]()) (Id = BTy_30 1);
		[Other](.) (Id = BTy_32 1);
		[String] (Id = BTy_33 1);
	End Pstruct;
End Punion

Refined Ty:
Punion(Id = BTy_25 886)
	Pstruct(Id = BTy_24 885)
		Poption(Id = BTy_1 885)
			Pstruct(Id = BTy_21 9)
				[Enum] {[StringConst] "compat-gcc-32-c", [StringConst] "compat-libstdc", [StringConst] "gcc-c", [StringConst] "gcc4-c", [StringConst] "gtk", [StringConst] "libstdc", [StringConst] "memtest86", } (Id = BTy_2 9);
				Punion(Id = BTy_16 9)
					Pstruct(Id = BTy_15 8)
						[StringConst] "+" (Id = BTy_4 8);
						Punion(Id = BTy_12 8)
							Pstruct(Id = BTy_11 6)
								[StringConst] "+" (Id = BTy_5 6);
								[Pint] (Id = BTy_7 6);
								Poption(Id = BTy_9 6)
									[Int] [~3...~2] (Id = BTy_10 2);
								End Poption;
							End Pstruct;
							[Int] [~1...~1] (Id = BTy_13 2);
						End Punion;
						[StringConst] "." (Id = BTy_14 8);
					End Pstruct;
					RArray(Id = BTy_20 1)
						[Other](+) (Id = BTy_19 3);
					End RArray;
				End Punion;
			End Pstruct;
		End Poption;
		[Host] (Id = BTy_22 885);
		[StringConst] "m" (Id = BTy_23 885);
	End Pstruct;
	Pstruct(Id = BTy_34 1)
		[String] (Id = BTy_26 1);
		[Other](.) (Id = BTy_27 1);
		[Other](() (Id = BTy_28 1);
		[String] (Id = BTy_29 1);
		[Other]()) (Id = BTy_30 1);
		[Other](.) (Id = BTy_32 1);
		[String] (Id = BTy_33 1);
	End Pstruct;
End Punion

====== Timing information ======
Tokenization time = 0.045
Measure1 time = 0.032
Reduce1 time = 1.057
Reduce2 time = 0.901
Reduce3 time = 0.602
Measure2 time = 0.025
================================

Outputing partitions to directory: gen/
Opening: gen/Ty, length 6
opening: gen/Complexity, length 14
opening: gen/Timing, length 10
Excutable directory:/Users/kzhu/pads/learning/golden
copy command: cp /Users/kzhu/pads/learning/golden/GNUMakefile.output gen/GNUmakefile

Completed data/rpmpkgs.txt
