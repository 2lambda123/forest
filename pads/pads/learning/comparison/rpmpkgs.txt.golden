Pstruct(35, tc: 437.347b, ac: 111.459b, dc: 146.054b, raw: 583.401b)
        [StringME] "/[0-9a-zA-Z]+/" (35, tc: 5.044b, ac: 36.000b, dc: 36b, raw: 41.044b);
        [StringConst] "-" (35, tc: 11.044b, ac: 0b, dc: 0b, raw: 11.044b);
        [StringME] "/[0-9a-zA-Z.]+/" (35, tc: 5.044b, ac: 36.000b, dc: 36b, raw: 41.044b);
        [StringConst] "-" (35, tc: 11.044b, ac: 0b, dc: 0b, raw: 11.044b);
        [StringME] "/[0-9a-zA-Z.]+/" (35, tc: 5.044b, ac: 36.000b, dc: 36b, raw: 41.044b);
        [StringConst] "." (35, tc: 11.044b, ac: 0b, dc: 0b, raw: 11.044b);
        [Enum] {[StringConst] "noarch", [StringConst] "i386", [StringConst] "i586", [StringConst] "i686", [StringConst] "alpha", [StringConst] "sparc", [StringConst] "mips", [StringConst] "ppc", [StringConst] "m68k", [StringConst] "SGI", [StringConst] "(none)", } (35, tc: 351.992b, ac: 3.459b, dc: 38.054b, raw: 390.046b);
        [StringConst] ".rpm" (35, tc: 29.044b, ac: 0b, dc: 0b, raw: 29.044b);
End Pstruct

